core/shared/src/main/scala/zio/IO.scala

```diff
@@ -730,6 +730,18 @@ object IO {
    */
   val none: UIO[Option[Nothing]] = ZIO.none
 
+  /**
+   * @see See [[zio.ZIO.noneOrFail]]
+   */
+  def noneOrFail[E](o: Option[E]): IO[E, Unit] =
+    ZIO.noneOrFail(o)
+
+  /**
+   * @see See [[zio.ZIO.noneOrFailWith]]
+   */
+  def noneOrFailWith[E, O](o: Option[O], f: O => E): IO[E, Unit] =
```

Comment:
You could have variants for some of the other type aliases that specialized the `E` type to `Throwable`.